app/src/org/commcare/activities/QueryRequestActivity.java

```diff
@@ -116,56 +132,176 @@ private void buildPromptUI() {
+        } else {
+            inputView = setUpEditTextView(promptView, queryPrompt, promptId, userAnswers, isLastPrompt);
+        }
+        setUpBarCodeScanButton(promptView, promptId, queryPrompt);
+
+        promptsLayout.addView(promptView);
+        promptsBoxes.put(promptId, inputView);
+    }
+
+    private void setUpBarCodeScanButton(View promptView, String promptId, QueryPrompt queryPrompt) {
+        ImageView barcodeScannerView = promptView.findViewById(R.id.barcode_scanner);
+        barcodeScannerView.setVisibility(isBarcodeEnabled(queryPrompt) ? View.VISIBLE : View.INVISIBLE);
+        barcodeScannerView.setTag(promptId);
+        barcodeScannerView.setOnClickListener(v ->
+                callBarcodeScanIntent((String)v.getTag())
+        );
+    }
+
+    private Spinner setUpSpinnerView(View promptView, QueryPrompt queryPrompt,
+                                     Hashtable<String, String> userAnswers) {
+        Spinner promptSpinner = promptView.findViewById(R.id.prompt_spinner);
+        promptSpinner.setVisibility(View.VISIBLE);
+        promptView.findViewById(R.id.prompt_et).setVisibility(View.GONE);
+
+        promptSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                Vector<SelectChoice> choices = queryPrompt.getItemsetBinding().getChoices();
+                SelectChoice selectChoice = choices.get(position);
+                String oldAnswer = userAnswers.get(queryPrompt.getKey());
+                if(oldAnswer == null || !oldAnswer.contentEquals(selectChoice.getValue())) {
+                    remoteQuerySessionManager.answerUserPrompt(queryPrompt.getKey(), selectChoice.getValue());
+                    refreshUI();
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+
+            }
+        });
+
+        setSpinnerData(userAnswers, queryPrompt, promptSpinner);
+        return promptSpinner;
+    }
+
+    private void setSpinnerData(Hashtable<String, String> userAnswers, QueryPrompt queryPrompt, Spinner promptSpinner) {
+        remoteQuerySessionManager.populateItemSetChoices(queryPrompt);
+        Vector<SelectChoice> items = queryPrompt.getItemsetBinding().getChoices();
+        String answer = userAnswers.get(queryPrompt.getKey());
+        String[] choices = new String[items.size()];
+        int selectedPosition = -1;
+        for (int i = 0; i < items.size(); i++) {
+            SelectChoice item = items.get(i);
+            choices[i] = item.getLabelInnerText();
+            if(item.getValue().equals(answer)){
+                selectedPosition = i;
+            }
+        }
+
+        ArrayAdapter<String> adapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, choices);
+        promptSpinner.setAdapter(adapter);
```

Comment:
Does this trigger a "selection changed" event at all, or is it only in like 208 that this gets processed?